src/main/java/gyro/google/compute/BackendBucketCdnPolicy.java

```diff
@@ -69,10 +229,63 @@ public String primaryKey() {
+        if (backendBucketCdnPolicy.getCacheKeyPolicy() != null) {
+            BackendBucketCdnCacheKeyPolicy cacheKeyPolicy = newSubresource(BackendBucketCdnCacheKeyPolicy.class);
+            cacheKeyPolicy.copyFrom(backendBucketCdnPolicy.getCacheKeyPolicy());
+            setCacheKeyPolicy(cacheKeyPolicy);
+        }
+
+        setBypassCacheOnRequestHeader(null);
+        if (!backendBucketCdnPolicy.getBypassCacheOnRequestHeadersList().isEmpty()) {
+            setBypassCacheOnRequestHeader(backendBucketCdnPolicy.getBypassCacheOnRequestHeadersList().stream()
+                .map(o -> {
+                    BackendBucketCdnBypassCacheOnRequestHeader bypassCacheOnRequestHeader = newSubresource(BackendBucketCdnBypassCacheOnRequestHeader.class);
+                    bypassCacheOnRequestHeader.copyFrom(o);
+                    return bypassCacheOnRequestHeader;
+                })
+                .collect(Collectors.toList()));
+        }
+
+        setNegativeCaching(backendBucketCdnPolicy.getNegativeCaching());
+        setNegativeCachingPolicy(null);
+        if (!backendBucketCdnPolicy.getNegativeCachingPolicyList().isEmpty()) {
+            setNegativeCachingPolicy(backendBucketCdnPolicy.getNegativeCachingPolicyList().stream()
+                .map(o -> {
+                    BackendBucketCdnNegativeCachingPolicy negativeCachingPolicy = newSubresource(BackendBucketCdnNegativeCachingPolicy.class);
+                    negativeCachingPolicy.copyFrom(o);
+                    return negativeCachingPolicy;
+                })
+                .collect(Collectors.toList()));
+        }
+
+        setClientTtl(backendBucketCdnPolicy.getClientTtl());
+        setDefaultTtl(backendBucketCdnPolicy.getDefaultTtl());
+        setMaxTtl(backendBucketCdnPolicy.getMaxTtl());
+        setRequestCoalescing(backendBucketCdnPolicy.getRequestCoalescing());
+        setServeWhileStale(backendBucketCdnPolicy.getServeWhileStale());
     }
 
     com.google.cloud.compute.v1.BackendBucketCdnPolicy toBackendBucketCdnPolicy() {
         return com.google.cloud.compute.v1.BackendBucketCdnPolicy.newBuilder()
-            .setSignedUrlCacheMaxAgeSec(getSignedUrlMaxAge()).build();
+            .setSignedUrlCacheMaxAgeSec(getSignedUrlMaxAge())
+            .setCacheMode(getCacheMode())
+            .setCacheKeyPolicy(getCacheKeyPolicy() != null ? getCacheKeyPolicy().toBackendBucketCdnPolicyCacheKeyPolicy() : new BackendBucketCdnCacheKeyPolicy().toBackendBucketCdnPolicyCacheKeyPolicy())
+            .addAllBypassCacheOnRequestHeaders(getBypassCacheOnRequestHeader().stream()
+                .map(BackendBucketCdnBypassCacheOnRequestHeader::toBackendBucketCdnPolicyBypassCacheOnRequestHeader)
+                .collect(Collectors.toList()))
+            .setClientTtl(getClientTtl() != null ? getClientTtl() : 0)
+            .setDefaultTtl(getDefaultTtl() != null ? getDefaultTtl() : 0)
+            .setMaxTtl(getMaxTtl() != null ? getMaxTtl() : 0)
```

Comment:
If these values will default to 0, we should have the getters return it and add it to the docs.